pod/perlintro.pod

```diff
@@ -162,7 +162,17 @@ More detailed information about Perl syntax can be found in L<perlsyn>.
 
 =head2 Perl variable types
 
-Perl has three main variable types: scalars, arrays, and hashes.
+You may have noticed the strange character string <$name> above.  That
+denotes a variable, which is generally pronounced as "name".  The dollar
+sign is called a "sigil", and indicates that "name" is a variable.
+Unlike many other computer languages where variables are just strings of
+typically only alphabetic and numeric characters, all Perl variables are
+preceded by a sigil.
+
+Perl has three main variable types: scalars, arrays, and hashes.  The
+particular sigil used for a variable name indicates the type of that
+variable.  We've already seen the dollar sign used as a sigil, C<$name>.
```

Comment:
I like the idea. I think it would be good to also introduce early here that the sigil can represent the type of data returned and not the variable type itself, as this is usually the most confusing aspect to newcomers. Perhaps: "indicates the type of that variable, or the type of data being accessed from the variable." And later, "The dollar sign indicates this variable is a scalar, or a scalar value is being accessed from another variable type."

For the other two sigils perhaps a small note like "The `@` sigil can also be used to access a slice of an array or hash." "The `%` sigil can also be used to access a key/value slice of an array or hash."

Reply:
Or maybe better to be even more specific about the dollar sigil. "The dollar sign indicates this variable is a scalar. In other cases, if a variable name is immediately followed by square brackets `[]` or braces `{}`, the dollar sign indicates accessing a scalar value of an array or hash respectively."

Reply:
That might be better to elaborate on in the section scalars. What do you think?

Reply:
Reading further down, it appeared to me that it was more natural to introduce your points in the next sections.